File: python/acronym/acronym.py
def abbreviate(words: str) -> str:
    filtered = list(filter(None, remove_delimiters(words).split(' ')))
    return ''.join([x[0].upper() for x in filtered])


def remove_delimiters(words: str) -> str:
    words = words.replace("-", " ")
    words = words.replace("_", " ")
    return words

File: ruby/acronym/acronym.rb
class Acronym
  def self.abbreviate(phrase)
    remove_delimiters(phrase)
      .split(' ')
      .map { |word| word[0].upcase }
      .join
  end

  def self.remove_delimiters(phrase)
    phrase.gsub!('-', ' ') if phrase.include?('-')
    phrase
  end
end
